noir_stdlib/src/embedded_curve_ops.nr

```diff
@@ -139,33 +139,42 @@ pub fn embedded_curve_add(
     point2: EmbeddedCurvePoint,
 ) -> EmbeddedCurvePoint {
     // docs:end:embedded_curve_add
-    let x_coordinates_match = point1.x == point2.x;
-    let y_coordinates_match = point1.y == point2.y;
-    let double_predicate = (x_coordinates_match & y_coordinates_match);
-    let infinity_predicate = (x_coordinates_match & !y_coordinates_match);
-    let point1_1 = EmbeddedCurvePoint {
-        x: point1.x + (x_coordinates_match as Field),
-        y: point1.y,
-        is_infinite: x_coordinates_match,
-    };
-    // point1_1 is guaranteed to have a different abscissa than point2
-    let mut result = embedded_curve_add_unsafe(point1_1, point2);
-    result.is_infinite = x_coordinates_match;
-
-    // dbl if x_match, y_match
-    let double = embedded_curve_add_unsafe(point1, point1);
-    result = if double_predicate { double } else { result };
-
-    // infinity if x_match, !y_match
-    if point1.is_infinite {
-        result = point2;
-    }
-    if point2.is_infinite {
-        result = point1;
+    if crate::runtime::is_unconstrained() {
+        embedded_curve_add_unsafe(point1, point2)
+    } else {
+        // In a constrained context we need to do some black magic in order to satisfy the backend's
+        // expectations about the inputs to an `embedded_curve_add` opcode.
+        //
+        // TODO: document this better.
```

Comment:
nit: are we tracking this?

Reply:
The comments above this function look slightly out of date as well

Reply:
https://github.com/noir-lang/noir/issues/7773